sqswatcher/plugins/sge.py

```diff
@@ -19,28 +17,64 @@
 import socket
 import logging
 import shlex
+from sqswatcher.sqswatcher import HostRemovalError
+from sqswatcher.sqswatcher import QueryConfigError
 
 log = logging.getLogger(__name__)
 
-def __runSgeCommand(command):
-    log.debug(repr(command))
+
+def _is_host_configured(command, hostname):
+    _command = shlex.split(command)
+    log.debug(_command)
+    host_configured = False
+
+    try:
+        output = sub.Popen(_command,
+                           stdout=sub.PIPE,
+                           env=dict(
+                                    os.environ,
+                                    SGE_ROOT='/opt/sge',
+                                    PATH='/opt/sge/bin:/opt/sge/bin/lx-amd64:/bin:/usr/bin',
+                                    ),
+                           ).communicate()[0]
+    except:
+        log.error("Failed to run %s\n" % command)
+        raise QueryConfigError
+
+    if output is not None:
+        # Expected output
+        # ip-172-31-66-16.ec2.internal
+        # ip-172-31-74-69.ec2.internal
+        match = filter(lambda x: hostname in x.split(".")[0], output.split("\n"))
+
+        if len(match) > 0:
```

Comment:
Note: `len(filter(...))` [is not supported](https://stackoverflow.com/questions/24291604/find-the-length-of-a-filter-object-in-python-3/24291630) by python3. 

http://python-future.org/compatible_idioms.html#filter-ifilter